src/System.Management.Automation/engine/Utils.cs

```diff
@@ -478,13 +478,13 @@ internal static string GetWindowsPowerShellVersionFromRegistry()
         }
 #endif
 
-        internal static string DefaultPowerShellAppBase => GetApplicationBase(DefaultPowerShellShellID);
+        internal static string DefaultPowerShellAppBase => GetApplicationBase();
 
-        internal static string GetApplicationBase(string shellId)
+        internal static string GetApplicationBase()
         {
-            // Use the location of SMA.dll as the application base.
+            // Use the location of SMA.dll as the application base or fallback to process home.
             Assembly assembly = typeof(PSObject).Assembly;
-            return Path.GetDirectoryName(assembly.Location);
+            return Path.GetDirectoryName(assembly.Location ?? Environment.ProcessPath);
```

Comment:
```suggestion
            return Path.GetDirectoryName(assembly.Location ?? AppContext.BaseDirectory ?? Environment.ProcessPath);
```
@SteveL-MSFT Sorry if I confused you. First part is for PowerShell executable scenario. As for second,  if we look `AppContext.BaseDirectory` implementation it covers common host scenarios and it has Environment.ProcessPath only for AOTNative scenario so we should add this too as last resort.

Reply:
@iSazonov I see.